cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -447,17 +463,17 @@ func (scaleSet *ScaleSet) TemplateNodeInfo() (*schedulercache.NodeInfo, error) {
 
 // Nodes returns a list of all nodes that belong to this node group.
 func (scaleSet *ScaleSet) Nodes() ([]cloudprovider.Instance, error) {
+	scaleSet.mutex.Lock()
+	defer scaleSet.mutex.Unlock()
+
 	vms, err := scaleSet.GetScaleSetVms()
 	if err != nil {
 		return nil, err
 	}
 
 	instances := make([]cloudprovider.Instance, 0, len(vms))
 	for i := range vms {
-		if len(*vms[i].ID) == 0 {
-			continue
-		}
-		name := "azure://" + *vms[i].ID
+		name := "azure://" + vms[i]
```

Comment:
okay, so we are moving the check which was here to one more layer below - right ?

Reply:
right